hooks/use-support-form.ts

```diff
@@ -0,0 +1,82 @@
+"use client";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import type { Address } from "viem";
+import { z } from "zod";
+import type { useHandleBuyFraction } from "./use-buy-fraction";
+
+interface SupportFormInputs {
+  fractionPayment: number;
+  comment: string;
+  hideName: boolean;
+  hideAmount: boolean;
+}
+
+const useSupportForm = (
+  dollarAmountNeeded: number,
+  pricePerUnit: number,
+  // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+  order: any,
+  handleBuyFraction: ReturnType<
+    typeof useHandleBuyFraction
+  >["handleBuyFraction"],
+  address: Address | undefined,
+  hypercertId: string | undefined
+) => {
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+
+  const fractionSaleFormSchema = z.object({
+    fractionPayment: z.coerce.number().min(1).max(Number(dollarAmountNeeded)),
+    comment: z.string(),
+    hideName: z.boolean(),
+    hideAmount: z.boolean(),
+  });
+
+  const form = useForm<SupportFormInputs>({
+    resolver: zodResolver(fractionSaleFormSchema),
+    defaultValues: {
+      fractionPayment: 1,
+      comment: "",
+      hideName: false,
+      hideAmount: false,
+    },
+  });
+
+  const onSubmit = async (values: SupportFormInputs) => {
+    setIsProcessing(true);
+    form.reset();
+    const unitsToBuy = values.fractionPayment / pricePerUnit;
+    if (!address) {
+      throw new Error("No address found");
+    }
+    try {
+      const txnReceipt = await handleBuyFraction(order, unitsToBuy, address);
```

Comment:
we're sending units here, but the CMS stores the purchase amounts in USD.  we should stick to sending amounts in USD when writing to the CMS